pkg/querier/queryrange/split_by_interval_test.go

```diff
@@ -261,6 +261,58 @@ func Test_splitMetricQuery(t *testing.T) {
 			},
 			interval: 3 * time.Hour,
 		},
+
+		// step larger than split interval
+		{
+			input: &LokiRequest{
+				StartTs: time.Unix(0, 0),
+				EndTs:   time.Unix(25*3600, 0),
+				Step:    6 * 3600 * seconds,
+			},
+			expected: []queryrange.Request{
+				&LokiRequest{
+					StartTs: time.Unix(0, 0),
+					EndTs:   time.Unix(6*3600, 0),
+					Step:    6 * 3600 * seconds,
+				},
+				&LokiRequest{
+					StartTs: time.Unix(6*3600, 0),
+					EndTs:   time.Unix(12*3600, 0),
+					Step:    6 * 3600 * seconds,
+				},
+				&LokiRequest{
+					StartTs: time.Unix(12*3600, 0),
+					EndTs:   time.Unix(18*3600, 0),
+					Step:    6 * 3600 * seconds,
+				},
+				&LokiRequest{
+					StartTs: time.Unix(18*3600, 0),
+					EndTs:   time.Unix(24*3600, 0),
+					Step:    6 * 3600 * seconds,
+				},
+				&LokiRequest{
+					StartTs: time.Unix(24*3600, 0),
+					EndTs:   time.Unix(25*3600, 0),
+					Step:    6 * 3600 * seconds,
+				},
+			},
+			interval: 15 * time.Minute,
+		},
+		{
+			input: &LokiRequest{
+				StartTs: time.Unix(0, 0),
```

Comment:
Does it make sense to add a scenario where `StartTs` is already bigger than the step? Or equal to the step?

Reply:
We can add it, but we do not do any alignment, so it didn't seem necessary.